test/runtime/Policies.go

```diff
@@ -631,6 +647,37 @@ var _ = Describe("RuntimeValidatedPolicies", func() {
 		connectivityTest(allRequests, helpers.App2, helpers.Httpd1, true)
 	})
 
+	It("L3-Dependent L7 Egress", func() {
+		_, err := vm.PolicyImportAndWait(vm.GetFullPath(policiesL3DependentL7EgressJSON), helpers.HelperTimeout)
+		Expect(err).Should(BeNil(), "unable to import %s", policiesL3DependentL7EgressJSON)
+
+		endpointIDS, err := vm.GetEndpointsIds()
+		Expect(err).To(BeNil(), "Unable to get IDs of endpoints")
+
+		app3EndpointID, exists := endpointIDS[helpers.App3]
+		Expect(exists).To(BeTrue(), "Expected endpoint ID to exist for %s", helpers.App3)
+
+		connectivityTest(httpRequestsPublic, helpers.App3, helpers.Httpd1, true)
+		checkProxyStatistics(app3EndpointID, 2, 2, 0, 2, 2)
+
+		connectivityTest(httpRequestsPrivate, helpers.App3, helpers.Httpd1, false)
+		checkProxyStatistics(app3EndpointID, 2, 4, 2, 2, 2)
+
+		connectivityTest(httpRequestsPublic, helpers.App3, helpers.Httpd2, true)
+		checkProxyStatistics(app3EndpointID, 4, 6, 2, 4, 4)
+
+		connectivityTest(httpRequestsPrivate, helpers.App3, helpers.Httpd2, true)
+		checkProxyStatistics(app3EndpointID, 6, 8, 2, 6, 6)
+
+		By("Disabling all the policies. All should work")
```

Comment:
Should we test  the `PolicyDel` here? that it's already tested in other test. The main issue if we deleted the policy it'll be difficult to debug. If you think that can be deleted it'll be great, if you consider that any problem here I'm happy to keep it.

Reply:
Great point, agreed. Removed this section since `AfterEach` here deletes all policies.